dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLitmusPTX.java

```diff
@@ -0,0 +1,310 @@
+        return new IValue(new BigInteger(ctx.getText()), getArchPrecision());
+    }
+
+    @Override
+    public Object visitInstructionRow(LitmusPTXParser.InstructionRowContext ctx) {
+        for (int i = 0; i < threadCount; i++) {
+            mainThread = i;
+            visitInstruction(ctx.instruction(i));
+        }
+        return null;
+    }
+
+    @Override
+    public Object visitStoreConstant(LitmusPTXParser.StoreConstantContext ctx) {
+        MemoryObject object = programBuilder.getOrNewObject(ctx.location().getText());
+        IValue constant = new IValue(new BigInteger(ctx.constant().getText()), getArchPrecision());
+        String mo = ctx.sem().content;
+        String scope;
+        switch (mo) {
+            case Tag.PTX.WEAK:
+                if (ctx.scope() != null) {
+                    throw new ParsingException("Weak store instruction doesn't need scope: " + ctx.scope().content);
+                }
+                scope = Tag.PTX.SYS;
+                break;
+            case Tag.PTX.REL:
+            case Tag.PTX.RLX:
+                scope = ctx.scope().content;
+                break;
+            default:
+                throw new ParsingException("Store instruction doesn't support sem: " + mo);
+        }
+        Store store = EventFactory.PTX.newTaggedStore(object, constant, mo, scope);
+        store.addFilters(ctx.store().storeProxy);
+        store.addFilters(Tag.PTX.CON);
+        return programBuilder.addChild(mainThread, store);
+    }
+
+    @Override
+    public Object visitStoreRegister(LitmusPTXParser.StoreRegisterContext ctx) {
+        MemoryObject object = programBuilder.getOrNewObject(ctx.location().getText());
+        Register register = programBuilder.getOrCreateRegister(mainThread, ctx.register().getText(), getArchPrecision());
+        String mo = ctx.sem().content;
+        String scope;
+        switch (mo) {
+            case Tag.PTX.WEAK:
+                if (ctx.scope() != null) {
+                    throw new ParsingException("Weak store instruction doesn't need scope: " + ctx.scope().content);
+                }
+                scope = Tag.PTX.SYS;
+                break;
+            case Tag.PTX.REL:
+            case Tag.PTX.RLX:
+                scope = ctx.scope().content;
+                break;
+            default:
+                throw new ParsingException("Store instruction doesn't support sem: " + mo);
+        }
+        Store store = EventFactory.PTX.newTaggedStore(object, register, mo, scope);
+        store.addFilters(ctx.store().storeProxy);
+        return programBuilder.addChild(mainThread, store);
+    }
+
+    @Override
+    public Object visitLocalConstant(LitmusPTXParser.LocalConstantContext ctx) {
+        Register register = programBuilder.getOrCreateRegister(mainThread, ctx.register().getText(), getArchPrecision());
+        IValue constant = new IValue(new BigInteger(ctx.constant().getText()), getArchPrecision());
```

Comment:
this can probably be replaced by `(IConst) ctx.constant().accept(this)`

Reply:
Replaced.